### DISCUSSION TITLE
Trouble Installing SP2 & Legacy Downloads for Vista Ultimate / Legal, Retail, activated with ID

### DISCUSSION THREAD

**Discussion by ghost** [LegacyUpdate/LegacyUpdate#100] [Questions & Help] [answered]
Is there any work around for failed installs? I downloaded (Twice / first "aborted") The second try, stupidly, I walked away while Service Pack 2 was loading only to discover that it stops working when the PC goes to sleep. I then set my PC to NEVER sleep, Never Go to screen saver and use full power settings. My third try, I played solitaire to keep PC awake, got the full download and then error message "Failed to in stall". It fails at the point where the downloaded update needs to "execute" the installation. 
![Legacy Update](https://user-images.githubusercontent.com/123204066/213790368-67fb4957-28d6-4dcd-835d-95a6e195b325.JPG)

**Comment by kirb** [comment]
Shoot, the failure to install is my silly mistake. It’s missing the .exe on the end of the filename. Easy fix, I’ll get one out ASAP. 

In the interim for your situation, you can keep Legacy Update sitting at the "Installation Aborted" screen, then type `%tmp%` into the Start search box. Open the nsXXXX.tmp folder, rename VistaSP2 to VistaSP2.exe, and double click it.

I'll see what I can do for the going to sleep issue. Windows definitely shouldn't go to sleep or put wifi in standby mode while it’s installing/downloading, that's not something I considered. There should be a pretty straightforward way I can ask Windows to stay awake while it’s running.

**Comment by ghost** [comment]
Dear Mr. Demasi,

Thank You very much for a really quick reply. I want to clarify, the last 2 downloads did not abort, they simply failed to install at the point where the "execute" should happen. I will try your instructions from the screen pictured below. Will that work?

Respectfully,
Jennifer Temple
________________________________

**Reply by kirb** [reply]
Yep, that will let you install the service pack manually. The problem is that it can't run because Windows doesn't know what it's supposed to do with the file, it's missing the .exe extension.

**Reply by ghost** [reply]
Thank You! This is the most useful service out there! It passed ALL of the virus/scam testing websites with better than 95% saying clan and safe. I have 4 PCs all running different OSs for various reasons BUT vista runs my legacy software perfectly! Look forward to my signing up to support your work. (Right after I finish dinner -p Hubby7 is yelling!)

**Reply by kirb** [reply]
I'm so glad Legacy Update is useful to you! No funny business - I just do this for the fun of it, and the income is from GitHub Sponsors/Patreon and the website ads.

I just released version 1.6.1 to fix both issues. Would really appreciate if you could give it a test! https://github.com/kirb/LegacyUpdate/discussions/102

**Reply by ghost** [reply]
Well, that did not go so well. I need to look up the error code first but I tried the new version on the website aqnd when it aborted I tried what you outlined earlier. The result is in the picture.
![Temp Files](https://user-images.githubusercontent.com/123204066/213841744-a7200689-4645-4771-8633-749ff93b4232.JPG)
![SP Fail](https://user-images.githubusercontent.com/123204066/213841752-42c536dd-4518-48eb-887c-b8aaf1307bc3.JPG)

**Reply by ghost** [reply]
P.S. I do not see error code 183 in the list of error codes on the legacydownload website.

**Reply by kirb** [reply]
Hmmm let me have a look and get back to you on that. Sorry it didn’t work!

One of those “nsXXXX.tmp” folders should have VistaSP2.exe in it that you can double click to install the service pack. Again keep the Legacy Update window open to the “Installation Aborted” screen while you do this - it’ll delete the folder if you click Cancel/X.

**Comment by ghost** [comment, marked as answer]
Dear Adam,

you should be able to tell from the screen shot that I did add the .exe to the ns temp file and started the install from double clicking there. (Only the first one of the ns temp folders was not an empty folder.) I am about to take another run at this. I'll keep you posted.

________________________________

**Reply by ghost** [reply]
VistaSP2.exe still fails to install, same error code 183, with a new twist
![No Luck SP2](https://user-images.githubusercontent.com/123204066/213874191-afcac5b1-e908-4a71-a048-4b78936876c3.JPG)

**Reply by ghost** [reply]
Could it be that my system lacks the .net 4 framework. I need to hunt around a download suitable for vista and see if fixes the legacyupdate problem.

**Reply by ghost** [reply]
I think we have a winner in 1.6.2, I just downloaded it, it downloaded well enough and went into "preparing your computer", can take an hour or more and the computer will restart several times. So, I will report back if the installation goes through successfully. "restarting windows update server" at this moment.

**Reply by ghost** [reply]
"Checking for latest updates. This could take a few minutes"

**Reply by ghost** [reply]
It LOOKS like SP2 installed & agent installed. I have left that PC open to the legacy.net page where .net 4 is installing now AND a slew of updates is showing. This is all looking VERY GOOD! .NET 4 completed, no problem. 218 ESSENTIAL updates now being downloaded and installed.

**Reply by ghost** [reply]
It finished running and restarting now. "Some updates were not installed". (only 3 failed)

**Reply by ghost** [reply]
Worked really well. I have a somewhat unrelated question. I see legacy Update uses Internet explorer by default. Would it work just as well from Firefox?

**Reply by kirb** [reply]
Great, so glad 1.6.2 worked! Sorry, I forgot to let you know it was released. 

IE is required because it uses a proprietary IE feature, ActiveX, which allows the website to access native functionality on your system. You could use IETab to access it in Firefox (if you can find a compatible version of it, not sure how far that supports).

Since you're on Vista, you can also use the Windows Update control panel if you'd prefer.

**Reply by ghost** [reply]
That's OK. I'll stay with what worked. If I run legacy again will it bring back all of the optional updates that I did not install the first time? BTW: By now you should be able to see getting a monthly coffee. Thank you for a great service!!

**Reply by ghost** [reply]
The updater fails to draw up any updates on a second run. I let it run all night! I thought I would look at and install some of the software updates it showed on the last run. I tried removing the updater and reinstalling to see if that would generate the list. I looked at hidden updates in control panel / updates and there are none hidden there. How might I get the updater to draw up the rest of the uninstalled updates? Home, install and hidden  on the legacyupdate.net are all grayed out and it goes straight to searching for updates but never finds any. (I'm on my third try today)

**Reply by kristibektashi** [reply]
> The updater fails to draw up any updates on a second run. I let it run all night! I thought I would look at and install some of the software updates it showed on the last run. I tried removing the updater and reinstalling to see if that would generate the list. I looked at hidden updates in control panel / updates and there are none hidden there. How might I get the updater to draw up the rest of the uninstalled updates? Home, install and hidden on the legacyupdate.net are all grayed out and it goes straight to searching for updates but never finds any. (I'm on my third try today)

I know this is an old thread, but I believe this issue has been fixed on Legacy Update v1.9

**Comment by AntonFromSweden** [comment]
This may be resolved now, but as far as I know, as a general comment, you cannot install Service Pack 2 (KB948465) for Windows Vista without having installed Service Pack 1 (KB936330)

Further information: https://support.microsoft.com/en-gb/topic/information-about-service-pack-2-for-windows-vista-and-for-windows-server-2008-7ddc6fc7-5809-45f9-4003-c351c5a2224e.
